code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm

```diff
@@ -646,15 +651,46 @@
 	reagent_state = LIQUID
 	color = "#800080"
 	strength = 5
+	overdose = 5  //5 units per ghostmushroom.
+	od_minimum_dose = 1
 	taste_description = "acid"
-	metabolism = REM
+	metabolism = REM * 0.5
 	unaffected_species = IS_DIONA | IS_MACHINE
+	var/datum/modifier/modifier
 
 /datum/reagent/toxin/spectrocybin/affect_blood(var/mob/living/carbon/M, var/removed)
 	..()
-	M.hallucination = max(M.hallucination, 50)
-	if(prob(10))
-		M.see_invisible = SEE_INVISIBLE_CULT
+	if(!is_overdosed)	
```

Comment:
Here, for example, just do:

`if(!dose > overdose)`, if needed swap dose with volume or w/e